hack/cluster-openshift.sh

```diff
@@ -538,6 +568,45 @@ if [ "$_CMD" = "up" ]; then
     KIALI_PASSPHRASE=${KIALI_PASSPHRASE} /tmp/deploy-kiali-to-openshift.sh
   fi
 
+  if [ "${KNATIVE_ENABLED}" == "true" ]; then
+    echo "Setting up security policy for knative..."
+
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-scc-to-user anyuid -z build-controller -n knative-build
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-scc-to-user anyuid -z controller -n knative-serving
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-scc-to-user anyuid -z autoscaler -n knative-serving
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-scc-to-user anyuid -z kube-state-metrics -n knative-monitoring
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-scc-to-user anyuid -z node-exporter -n knative-monitoring
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-scc-to-user anyuid -z prometheus-system -n knative-monitoring
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-cluster-role-to-user cluster-admin -z build-controller -n knative-build
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-cluster-role-to-user cluster-admin -z controller -n knative-serving
+
+    curl -L https://github.com/knative/serving/releases/download/v0.4.1/serving.yaml \
+      | sed 's/LoadBalancer/NodePort/' \
+      | ${MAISTRA_ISTIO_OC_COMMAND} apply -f -
+
+    echo "Waiting for Knative to become ready"
+    sleep 5; while echo && ${MAISTRA_ISTIO_OC_COMMAND} get pods -n knative-serving | grep -v -E "(Running|Completed|STATUS)"; do sleep 5; done
+
+    echo "Creating a new project for knative examples"
+    ${MAISTRA_ISTIO_OC_COMMAND} new-project knative-examples || true
+
+    echo "Applying default domain to knative pods"
+
+    export DOMAIN=$(${MAISTRA_ISTIO_OC_COMMAND} get route -n istio-system istio-ingressgateway --output=custom-columns=ROUTE:.spec.host | grep -v ROUTE | sed "s/istio-ingressgateway-istio-system.//g")
+    cat ${SCRIPT_ROOT}/knative/config-domain.yaml | envsubst | oc apply -f -
+
+    echo "Using domain: *.knative.${DOMAIN}"
+
+    ${MAISTRA_ISTIO_OC_COMMAND} adm policy add-scc-to-user privileged -z default -n knative-examples
+    ${MAISTRA_ISTIO_OC_COMMAND} label --overwrite namespace knative-examples istio-injection=enabled
+
+    echo "Knative is installed!"
+
+    echo "Installing a sample application for knative..."
+    istiooc delete -n knative-examples -f ${SCRIPT_ROOT}/knative/service.yaml || true
+    istiooc apply -n knative-examples -f ${ROOT}/knative/service.yaml
```

Comment:
What is ROOT? Where is this defined?

Reply:
Whoops, did a late replace on the commit, let me change :)

Reply:
Fixed.